app/pages/lab/actions/workflow.js

```diff
@@ -1,14 +1,27 @@
 var apiClient = require('panoptes-client/lib/api-client');
 
+var projects = apiClient.type('projects');
 var workflows = apiClient.type('workflows');
 
 var workflowActions = {
-  createWorkflowForProject: function(projectID, workflowData) {
+  createWorkflowForProject: function(project, workflowData) {
     var allWorkflowData = Object.assign({
-      links: {project: projectID}
+      links: {project: project.id}
     }, workflowData);
 
     return workflows.create(allWorkflowData).save();
+  },
+  copyWorkflowForProject: async function(project, workflowData) {
+    const oldLinks = new Set(project.links.workflows);
+    const [newProject] = await project.addLink('workflows', [workflowData.id]);
+    const newLinks = newProject?.links
```

Comment:
not blocking but maybe for consistency of the links lookup and variable naming
```suggestion
    const newLinks = newProject?.links.workflows
```